### PR TITLE
Converts build-client.yml pipeline to 1ES template

### PR DESCRIPTION
**Pull request by seanimam**
## Description

This PR converts build-client.yml to the new 1ES template required by Microsoft with the minimal required set of template changes required. This new template enables security and auditing checks by Microsoft. Read more on how the logic for converting to a 1ES pipeline works on [the official 1ES wiki](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/1es-pipeline-templates/onboarding/overview)

A temporary 1ES folder was created for converted pipeline templates. This enables other pipelines to remain unchanged and use the original templates while new PR's are opened for individually converted other pipelines to 1ES as required by Microsoft.

- Once all pipelines are moved over, we can start moving the files out of the 1ES folder and eventually remove the folder since it will no longer have any files (This is after we convert the rest of the required pipelines). 

## Reviewer Guidance

- Confirm the newly converted pipelines run successfully. Look at the completed steps and compare with a recent successful run on main to ensure the expected steps run (plus the new SDL and 1ES security checks)
- Make sure all the relevant templates and pipelines for  build-client.yml have been converted
- Make sure there are no unnecessary changes or 1ES template and/or pipeline conversions
- Make sure the yml is valid -- a successful pipeline run should suffice for confirming this

### CODE REVIEW

**Review by alexvy86 — COMMENTED** [review]
The changes to build-client are fine. However, when we merge this our main pipeline is going to start taking a noticeable amount of additional time to run, so I think we should update `timeoutInMinutes` in build-npm-package to 90 in this PR and once it's merged monitor how long pipeline runs take to complete (both for PRs and CI), to see if we can adjust that number down.

**Review by alexvy86 — COMMENTED** [review]
Two changes I noticed went into build-npm-package in main but we haven't ported to the 1ES version: 
- https://github.com/microsoft/FluidFramework/pull/18867/files#diff-b1f37b84a08cf91822f3ded431e0b08d7226f6bba02275e178ba65ff904ebb15
- https://github.com/microsoft/FluidFramework/pull/18788/files#diff-b1f37b84a08cf91822f3ded431e0b08d7226f6bba02275e178ba65ff904ebb15

I think those would be the last and we're good to merge.

**Comment by seanimam** [comment]
> Two changes I noticed went into build-npm-package in main but we haven't ported to the 1ES version:
> 
> * https://github.com/microsoft/FluidFramework/pull/18867/files#diff-b1f37b84a08cf91822f3ded431e0b08d7226f6bba02275e178ba65ff904ebb15
> * https://github.com/microsoft/FluidFramework/pull/18788/files#diff-b1f37b84a08cf91822f3ded431e0b08d7226f6bba02275e178ba65ff904ebb15
> 
> I think those would be the last and we're good to merge.

Great callouts. Changes have been added to both PR's

File: tools/pipelines/templates/1ES/build-npm-package.yml
Comment (alexvy86) [inline tools/pipelines/templates/1ES/build-npm-package.yml:199]: Note for reviewers: this is already in the original `build-npm-package.yml`, it's not a change required only in the 1ES version.

File: tools/pipelines/templates/1ES/build-npm-package.yml
Comment (alexvy86) [inline tools/pipelines/templates/1ES/build-npm-package.yml:633]: Note for reviewers: this is already in the original `build-npm-package.yml`, just porting it to the 1ES version.

**Review by alexvy86 — APPROVED** [review]
Looks good. Just final call for when we merge this, we need to be super proactive in monitoring PR and CI runs to react quickly to any issues we might have missed.

**Pull request merged**
